### PR TITLE
Reddit Presence While Logged Out Fixed

### LINKED ISSUE #3711: [Presence | Bug] Reddit
**Describe the Bug**
 Reddit presence is not working. 

**Steps to Reproduce**

1. Go to https://www.reddit.com/
2. Make sure Premid application is working and the reddit presence is installed.
3. Go to ur activity status and it doesnt work.

**Expected Behavior**
 I was expecting the reddit presence to work.

**Screenshots**

https://imgur.com/a/cnO7prC
https://imgur.com/a/99xW2f3

**Environment**
 
- OS:  Windows 10 Pro x64
- Browser:  Brave 
- Version:  Brave  v1.24.84  (x64)

### PR DESCRIPTION
**Pull request by DarkVillager**
Resolves #3711
image
image

### CODE REVIEW

File: websites/R/Reddit/presence.ts [hunk 1]
```diff
@@ -83,13 +83,23 @@ presence.on("UpdateData", async () => {
...
+      );
+      if (sub == null || !sub) {
+       sub = document.querySelector("#SHORTCUT_FOCUSABLE_DIV > div:nth-child(4) > div > div > div > div._3ozFtOe6WpJEMUtxDOIvtU > div.q4a8asWOWdfdniAbgNhMh > div > div.QscnL9OySMkHhGudEvEya > div > div._3TG57N4WQtubLLo8SbAXVF > h2");
+       if (!sub) {
+        presenceData.details = (await strings).browsing;
+        presenceData.state = "Home";
+       } else {
+       presenceData.details = (await strings).browsing;
+       presenceData.state = sub.textContent;
+       }
```
Comment (ririxidev) [inline websites/R/Reddit/presence.ts:98]:
```suggestion
        presenceData.details = (await strings).browsing;
        presenceData.state = !sub ? "Home" : sub.textContent;
```

File: websites/R/Reddit/presence.ts [hunk 2]
```diff
@@ -83,13 +83,23 @@ presence.on("UpdateData", async () => {
       presenceData.smallImageKey = "live";
       presenceData.smallImageText = (await strings).live;
     } else {
-      subReddit = document.querySelector(
+      delete presenceData.smallImageKey
+      let sub = document.querySelector(
```
Comment (ririxidev) [inline websites/R/Reddit/presence.ts:87]:
```suggestion
      const sub = document.querySelector(
```

File: websites/R/Reddit/presence.ts [hunk 3]
```diff
@@ -83,13 +83,23 @@ presence.on("UpdateData", async () => {
...
         "span._1GieMuLljOrqnVpRAwz7VP"
-      ).textContent;
+      );
+      if (sub == null || !sub) {
+       sub = document.querySelector("#SHORTCUT_FOCUSABLE_DIV > div:nth-child(4) > div > div > div > div._3ozFtOe6WpJEMUtxDOIvtU > div.q4a8asWOWdfdniAbgNhMh > div > div.QscnL9OySMkHhGudEvEya > div > div._3TG57N4WQtubLLo8SbAXVF > h2");
+       if (!sub) {
```
Comment (ririxidev) [inline websites/R/Reddit/presence.ts:92]: wait - what's the point of these ifs

**Review by ririxidev — CHANGES_REQUESTED** [review]

File: websites/R/Reddit/presence.ts (hunk 3 above)
Comment (DarkVillager) [reply websites/R/Reddit/presence.ts]: I reassigned sub so there are two different elements thats why multiple ifs looks weird tho

Comment (ririxidev) [reply websites/R/Reddit/presence.ts]: makes sense but why if sub == null || !sub bruh

Comment (DarkVillager) [reply websites/R/Reddit/presence.ts]: cause idk

File: websites/R/Reddit/presence.ts [hunk 4]
```diff
@@ -83,13 +83,18 @@ presence.on("UpdateData", async () => {
       presenceData.smallImageKey = "live";
       presenceData.smallImageText = (await strings).live;
     } else {
-      subReddit = document.querySelector(
+      delete presenceData.smallImageKey;
+      const sub = document.querySelector(
         "span._1GieMuLljOrqnVpRAwz7VP"
-      ).textContent;
+      );
+      if (sub == null || !sub) {
```
Comment (ririxidev) [inline websites/R/Reddit/presence.ts:90]:
```suggestion
      if (sub === null) {
```

**Review by ririxidev — APPROVED** [review]

File: websites/R/Reddit/presence.ts [hunk 5]
```diff
@@ -83,13 +83,18 @@ presence.on("UpdateData", async () => {
...
+       sub = document.querySelector("#SHORTCUT_FOCUSABLE_DIV > div:nth-child(4) > div > div > div > div._3ozFtOe6WpJEMUtxDOIvtU > div.q4a8asWOWdfdniAbgNhMh > div > div.QscnL9OySMkHhGudEvEya > div > div._3TG57N4WQtubLLo8SbAXVF > h2");
+        presenceData.details = (await strings).browsing;
+        presenceData.state = !sub ? "Home" : sub.textContent;
+      } else {
       presenceData.details = (await strings).browsing;
-      presenceData.state = subReddit;
-      delete presenceData.smallImageKey;
-    }
+      presenceData.state = sub.textContent;
+    } }
```
Comment (Bas950) [inline websites/R/Reddit/presence.ts:97]: is this double close correct?
cant really see it because you didnt prettify it..

Comment (DarkVillager) [reply websites/R/Reddit/presence.ts:97]: yes

**Comment by ririxidev** [comment]
run prettier anyway

**Review by ririxidev — APPROVED** [review]

**Review by Bas950 — APPROVED** [review]

**Pull request merged**